src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -3109,6 +3158,7 @@ internal Buffer(IEnumerable<TElement> source)
                     count++;
                 }
             }
+
             this.items = items;
```

Comment:
That's a fine separate question, but to be clear mine wasn't about dependencies. When I previously did a sweep to replace new array allocations with Array.Empty, I generally refrained from doing so in ToArray methods, as many are documented to allocate or create a new arrays, and I didn't want to inadvertently introduce compat problems for folks relying on the methods returning new instances. If we don't believe that's an issue in this case, I'd be happy to see it changed, too.